script/modelling/power_systems.jl

```diff
@@ -1,326 +1,329 @@
-
-#' For further reading on the UC problem we refer interested readers to G. Morales-Espana, J. M. Latorre, and A. Ramos, "Tight and Compact MILP Formulation for the Thermal Unit Commitment Problem," IEEE Transactions on Power Systems, vol. 28, pp. 4897-4908, 2013.
-#' In the following example we convert the ED model explained above to the UC model. 
-
-# In this cell we introduce binary decision u to the economic dispatch problem (function solve_ed)
-function solve_uc(g_max, g_min, c_g, c_w, d, w_f)
-    #Define the unit commitment (UC) model
-    uc = Model(with_optimizer(GLPK.Optimizer))
-    
-    # Define decision variables    
-    @variable(uc, 0 <= g[i=1:2] <= g_max[i]) # power output of generators
-    @variable(uc, u[i = 1:2], Bin) # Binary status of generators
-    @variable(uc, 0 <= w <= w_f ) # wind power injection
-
-    # Define the objective function
-    @objective(uc, Min, dot(c_g, g) + c_w * w)
-
-    # Define the constraint on the maximum and minimum power output of each generator
-    @constraint(uc, [i = 1:2], g[i] <= g_max[i]) #maximum
-    @constraint(uc, [i = 1:2], g[i] >= g_min[i]) #minimum
-
-    # Define the constraint on the wind power injection
-    @constraint(uc, w <= w_f)
-
-    # Define the power balance constraint
-        @constraint(uc, sum(g) + w == d)
-
-    # Solve statement
-    optimize!(uc)
-    
-    status = termination_status(uc)
-    return status, value.(g), value(w), w_f - value(w), value.(u), objective_value(uc)
-end
-
-# Solve the economic dispatch problem
-status, g_opt, w_opt, ws_opt, u_opt, obj = solve_uc(g_max, g_min, c_g, c_w, d, w_f);
-
-println("\n")
-println("Dispatch of Generators: ", g_opt[:], " MW")
-println("Commitments of Generators: ", u_opt[:])
-println("Dispatch of Wind: ", w_opt, " MW")
-println("Wind spillage: ", w_f - w_opt, " MW") 
-println("\n")
-println("Total cost: ", obj, "\$")
- 
-#' ### Unit Commitment as a function of demand
- 
-#' After implementing the UC model, we can now assess the interplay between the minimum power output constraints on generators and wind generation.
-
-uc_df = DataFrame(Symbol("Commitment of Generator 1(MW)") => Float64[],
-               Symbol("Commitment of Generator 2(MW)") => Float64[],
-               Symbol("Dispatch of Generator 1(MW)") => Float64[],
-               Symbol("Dispatch of Generator 2(MW)") => Float64[],
-               Symbol("Dispatch of Wind(MW)") => Float64[],
-               Symbol("Spillage of Wind(MW)") => Float64[],
-               Symbol("Total cost(\$)") => Float64[])
-
-for demandscale = 0.2:0.1:1.5
-    status, g_opt, w_opt, ws_opt, u_opt, obj = solve_uc(g_max, g_min, c_g, c_w, demandscale*d, w_f)
- 
-    if status == MOI.OPTIMAL
-    push!(uc_df, (u_opt[1], u_opt[2], g_opt[1], g_opt[2], w_opt, ws_opt, obj))
-    else
-        println("Status: $status for demandscale = $demandscale \n")
-    end
-end
-
-#+
-
+#' ---


```

Comment:
Line endings in this file got borked?

Reply:
arg this one got me, thanks

Reply:
I'm not sure which tool is messing up with line endings